filament/src/VertexBuffer.cpp

```diff
@@ -188,6 +196,7 @@ size_t FVertexBuffer::getVertexCount() const noexcept {
 
 void FVertexBuffer::setBufferAt(FEngine& engine, uint8_t bufferIndex,
         backend::BufferDescriptor&& buffer, uint32_t byteOffset) {
+    ASSERT_PRECONDITION_NON_FATAL(!mBufferObjectsEnabled, "Please use setBufferObjectAt()");
```

Comment:
You cannot (shouldn't) use `ASSERT_PRECONDITION_NON_FATAL` here because you cannot return the failure to the user in any meaningful way. Use `ASSERT_PRECONDITION` instead.